src/libraries/System.Reflection/System.Reflection.sln

```diff
@@ -1,4 +1,8 @@
-﻿Microsoft Visual Studio Solution File, Format Version 12.00
```

Comment:
Why? Is the "Checked" sorting not stable?

Reply:
p.s. in the past I've manually applied changes to the .sln, but it seems like the sorting VS wants to do now is stable and based on sorting "Checked" before Debug\Release and also incorrect guid sorting which was likely due to manual editing.

Reply:
We use slngen in dotnet/runtime to automatically generate/update the solution files for all the libraries. These generated solution files are slightly different to the ones that VS emits (when saving the solution file inside the IDE). As an example, the header which contains the VS version string is missing from the slngen generated ones because we don't want that encoded in the repo.

You can for sure check these changes in but they don't add any value from what I see and the slns will be overwritten the next time when all solution files are updated via slngen.

Reply:
I'll check in as-is since modifying the .sln manually is error prone.

If the slngen tool doesn't match VS for sorting "Checked" and by project GUID then it should be updated so there are minimal diffs between VS.

Reply:
Update: there was a conflict; I just pull the latest changes which did some other clean-up. So this PR doesn't change the .sln any longer.